Add guidance for low quality or low engagement

JimMadge opened the pull request:
Closes #4424 

- Adds guidance for low quality and low engagement contributions to contribution guidelines
  - Explains some examples or criteria
  - Explains that contributions deemed low quality may be rejected
- Add item to gardening guidelines to close low quality contributions (referring back to contribution guidelines)

JimMadge committed:
Use glossary more in contribution guide

JimMadge committed:
Add low-effort section to contributing guidlines

JimMadge committed:
Refer to low-effort contributions in gardening guidelines

JimMadge committed:
Merge remote-tracking branch 'upstream/main' into low-effort

JimMadge committed:
Add emphasis

JimMadge committed:
Restructure examples

JimMadge committed:
Remove empty line

Arielle-Bennett commented on book/website/community-handbook/contributing.md:
```diff
@@ -108,33 +104,65 @@ You can also read the [Authoring section of the MyST Markdown Guide](xref:myst-g
   - Open a draft pull request if the work is still in progress
-  - Open a pull request if you feel the work is ready for review
+  - Open a pull request if you feel the work is ready for {term}`review`
 4. Iterate on changes in response to CI and reviewer comments
```
Just caught that "CI" is jargon and probably needs to be added to the glossary 😩

EstherPlomp reviewed (approved):
I think this is great! I have nothing to add! Thanks @JimMadge

Arielle-Bennett commented on book/website/community-handbook/contributing.md:
```diff
@@ -108,33 +104,65 @@ You can also read the [Authoring section of the MyST Markdown Guide](xref:myst-g
+And so, pull requests may be closed if we feel they are unlikely to result in a _tangible improvement_ to the book, or the maintainer investment required would not match the value of the contribution.
+
+:::{important}
+There is no size threshold contributions to be accepted.
```
```suggestion
There is no size threshold for contributions to be accepted.
```

Arielle-Bennett commented on book/website/community-handbook/contributing.md:
```diff
@@ -108,33 +104,65 @@ You can also read the [Authoring section of the MyST Markdown Guide](xref:myst-g
+- Apparently machine-generated contributions with little or no human effort to review and edit.
+  This may be apparent in comments as:
+    - Irrelevant, unclear text.
+    - Disconnection between {term}`pull request` comments and changes.
```
```suggestion
    - Disconnection between {term}`pull request` comments describing changes and actual changes made to the  files.
```

Arielle-Bennett commented on book/website/community-handbook/contributing.md:
```diff
@@ -108,33 +104,65 @@ You can also read the [Authoring section of the MyST Markdown Guide](xref:myst-g
+
+- Not responding to reviewers.
+- Ignoring reviewers and instead making unrelated changes.
+
```
```suggestion
- Not using project templates for {term}`issues` or pull requests.
```

JimMadge replied:
This might be a bit more contentious, I often skip the templates if I think I know what to write or the changes are clear (or I'm feeling lazy 😅)

Arielle-Bennett replied:
That works! I've just noticed that most AI stuff comes in without a proper PR template completed which is a dead giveaway - but I agree we don't want to be super rigid if people have made an effort...

JimMadge committed:
Update book/website/community-handbook/contributing.md

Co-authored-by: Arielle-Bennett <74651964+Arielle-Bennett@users.noreply.github.com>

JimMadge committed:
Update book/website/community-handbook/contributing.md

Co-authored-by: Arielle-Bennett <74651964+Arielle-Bennett@users.noreply.github.com>

JimMadge committed:
Add references to guiding principles

JimMadge committed:
Update references to CoC and contributing

JimMadge committed:
Merge remote-tracking branch 'origin/low-effort' into low-effort

JimMadge committed:
Update continuous integration definition

JimMadge committed:
Add note on pull request template

JimMadge committed:
Merge remote-tracking branch 'upstream/main' into low-effort

JimMadge committed:
Add bad issue/PR text as a low quality indicator

Arielle-Bennett commented on book/website/community-handbook/contributing.md:
```diff
@@ -74,28 +70,28 @@ You can also contact the Community Management Working Group and the Steering Com
 
 Look through open items first; join discussions, collaborate on an ongoing tasks and exchange your thoughts with others.
 If you can't find an open item related to your idea, you can open a new one.
 Open and discussion if you have a question, or an idea to talk about with the community.
```
```suggestion
Open a discussion if you have a question, or an idea to talk about with the community.
```

JimMadge committed:
Update book/website/community-handbook/contributing.md

Co-authored-by: Brigitta Sipőcz <b.sipocz@gmail.com>

JimMadge committed:
Update book/website/community-handbook/contributing.md

Co-authored-by: Brigitta Sipőcz <b.sipocz@gmail.com>

JimMadge committed:
Reword sentence for clarity

Arielle-Bennett commented on book/website/community-handbook/github-gardening.md:
```diff
@@ -165,6 +165,7 @@ This step keeps the repository clean, closing inactive or completed work while l
 #### Pruning
 
 - [ ] Merge if it is ready, passes tests and has accepted reviews
+- [ ] Close if the pull request is an example of [low quality or low engagement](#ch-contributing-low-effort)
```
Yes should go in the gardening guide!

JimMadge committed:
Move gardening chapter up in ToC

JimMadge committed:
Correct labels

Arielle-Bennett reviewed (commented):
Sorry, I keep meaning to finish this review :weary:

JimMadge committed:
Update book/website/community-handbook/github-gardening.md

Co-authored-by: Brigitta Sipőcz <b.sipocz@gmail.com>

JimMadge committed:
Apply suggestions from code review

Co-authored-by: Arielle-Bennett <74651964+Arielle-Bennett@users.noreply.github.com>

JimMadge committed:
Add response template/guidance

JimMadge committed:
Merge remote-tracking branch 'origin/low-effort' into low-effort

JimMadge committed:
Add checklist to PR template

JimMadge commented:
I have added some response fragments. I slightly changed what @EstherPlomp suggested (because the contributing guidelines don't actually say you _must_ declare AI use) but it is largely the same with a few additions.

I've also added (thanks to discussion with @sgibson91) a checklist to the PR template, prompting people to check the "important" things like CoC, contributing guide, style guide. That should help flag these things to contributors, so it doesn't feel unfair when we refer to them.

EstherPlomp reviewed (approved):
I'm happy with your changes @JimMadge - thanks!

I had to laugh about: " We also recommend that you remix the suggestions here in your own voice as it helps to show our personalities and prevent gardeners appearing robotic."  - and also yes :)

sgibson91 commented on book/website/afterword/glossary.md:
```diff
@@ -107,7 +107,10 @@ continuous deployment
-: It is the practice of integrating changes to a project made by individuals into a main, shared version frequently (usually multiple times per day). Also called CI.
+: A development practice where changes are automatically, and rapidly, integrated into the development branch.
+  This approach helps ensure the development code works and helps avoid large conflicts between feature branches.
+  The term is strongly associated with tools which support this way or working, such as automated building and testing, which are often called continuous integration tools.
```
```suggestion
  The term is strongly associated with tools which support this way of working, such as automated building and testing, which are often called continuous integration tools.
```

sgibson91 commented on book/website/community-handbook/contributing.md:
```diff
@@ -108,33 +104,68 @@ You can also read the [Authoring section of the MyST Markdown Guide](xref:myst-g
-  - Open a pull request if you feel the work is ready for review
-4. Iterate on changes in response to CI and reviewer comments
-5. Merge when all issues are addressed and changes are approved
+1. Create a feature branch on your fork.
```
Can we define the term "feature branch" (if not already defined, reviewing on my phone). It sounds like a magical, special branch if you don't know much about git, when really it's "just" a new branch

JimMadge committed:
Update book/website/afterword/glossary.md

Co-authored-by: Sarah Gibson <44771837+sgibson91@users.noreply.github.com>

JimMadge committed:
Add feature branch definition

JimMadge committed:
Update branch definition

JimMadge committed:
Merge branch 'main' into low-effort

JimMadge committed:
Merge remote-tracking branch 'origin/low-effort' into low-effort

Arielle-Bennett reviewed (approved):
Looking good to me! One small comment on requesting reviews which I think is harder now for contributors...

JimMadge committed:
Update book/website/community-handbook/github-gardening.md

Co-authored-by: Arielle-Bennett <74651964+Arielle-Bennett@users.noreply.github.com>
